docs: Updates docs to add mentions of `beta` (#44)
open-sauced/docs.opensauced.pizza: docs/technical/resolve-merge-conflicts.md

## Merge with upstream

> Then merge with the forked up-to-date default branch:

Reviewer: ```suggestion
Then merge with the forked up-to-date `HEAD` (default branch):
```

> Then merge with the forked up-to-date `HEAD` (default branch):

Reviewer: ```suggestion
Then merge with the forked up-to-date `beta` (default branch):
```

## Resolve conflicts

> Since this pull request does not modify the `package.json` file it is safe to fast forward the changes from `origin/insert-default-branch`:

Reviewer: on GitHub `HEAD` redirects to the default branch. This is also through for git. It would be confusing for new contributors to see `insert-default-branch` especially since most will lean towards copy and pasting. 

Replace `insert-default-branch` with `HEAD`. Also, put a note similar to what I mentioned about `HEAD` redirecting to the default branch, and I think that would cover the bases and make this mergeable.

However when that happens, it is advised to:
- fast-forward `npm-shrinkwrap.json` 

> - fast-forward deleted and modified `upstream/HEAD` changes to `package.json` 

Reviewer: ```suggestion
- fast-forward deleted and modified `upstream/beta` changes to `package.json` 
```

> - fast-forward deleted and modified `upstream/insert-default-branch` changes to `package.json` 

Reviewer: ```suggestion
- fast-forward deleted and modified `upstream/HEAD` changes to `package.json` 
```

- fast-forward your added lines to `package.json`

> - run `npm ci` to delete local modules and create dependency resolution from `upstream/HEAD`

Reviewer: ```suggestion
- run `npm ci` to delete local modules and create dependency resolution from `upstream/beta`
```

> - run `npm ci` to delete local modules and create dependency resolution from `upstream/insert-default-branch`

Reviewer: ```suggestion
- run `npm ci` to delete local modules and create dependency resolution from `upstream/HEAD`
```
